pkg/apis/awsprovider/v1alpha1/awsmachineproviderconfig_types.go

```diff
@@ -72,6 +73,18 @@ type AWSMachineProviderSpec struct {
 	// KeyName is the name of the SSH key to install on the instance.
 	// +optional
 	KeyName string `json:"keyName,omitempty"`
+
+	// JoinConfiguration is used to customize any kubeadm join configuration
+	// parameters.
+	JoinConfiguration v1beta1.JoinConfiguration `json:"joinConfiguration,omitempty"`
```

Comment:
What do you think about namespacing these in a different `kubeadmConfiguration` struct?

Reply:
what do you mean by namespaced? The machine is already namespaced via metav1.Meta so I'm guessing you mean a different kind of namespace...?

Reply:
Sorry I meant in a sub-struct where the json key would be named `kubeadmConfiguration`

Reply:
Ahhhh, gotcha. Yeah, makes sense to group these

Reply:
I'd recommend that the key just be `kubeadm` and then that would allow for additional fields to be added later even if they aren't init/join related (overriding the kubeadm binary/version/etc) down the road.